scipy/sparse/csgraph/_matching.pyx

```diff
@@ -0,0 +1,266 @@
+cimport cython
+import numpy as np
+cimport numpy as np
+
+from scipy.sparse import (csr_matrix,
+                          isspmatrix_coo, isspmatrix_csc, isspmatrix_csr)
+
+
+def maximum_bipartite_matching(graph, perm_type='row'):
+    r"""
+    maximum_bipartite_matching(graph, perm_type='row')
+
+    Returns a matching of a bipartite graph whose cardinality is as least that
+    of any given matching of the graph.
+
+    Parameters
+    ----------
+    graph : sparse matrix
+        Input sparse in CSR format whose rows represent one partition of the
+        graph and whose columns represent the other partition. An edge between
+        two vertices is indicated by the corresponding entry in the matrix
+        existing in its sparse representation.
+    perm_type : str, {'row', 'column'}
+        Which partition to return the matching in terms of: If ``'row'``, the
+        function produces an array whose length is the number of columns in the
+        input, and whose :math:`j`'th element is the row matched to the
+        :math:`j`'th column. Conversely, if ``perm_type`` is ``'column'``, this
+        returns the columns matched to each row.
+
+    Returns
+    -------
+    perm : ndarray
+        A matching of the vertices in one of the two partitions. Unmatched
+        vertices are represented by a ``-1`` in the result.
+
+    Notes
+    -----
+    This function uses of the Hopcroft--Karp algorithm [1]_. Its time
+    complexity is :math:`O(\lvert E \rvert \sqrt{\lvert V \rvert})`, and its
+    space complexity is linear in the number of rows. In practice, this
+    asymmetry between rows and columns means that it can be more efficient to
+    transpose the input if it contains more columns than rows.
+
+    By Kőnig's theorem, the cardinality of the matching is also the number of
```

Comment:
Glancing over a few of the CI results, it looks like the source file `scipy/sparse/csgraph/_matching.py` has some unicode / encoding issues. Not sure if it just this character or another / others. Looks like some Windows and at least 1 Linux image don't like the source file for that reason. Locally my machine seems "ok" with it.

I think a solution might be to add `# -*- coding: utf-8 -*-` at the top of this source file, and I do remember that @larsoner wrote #9615 after we started having these kinds of issues in the past & indeed his CI guard is catching this with a `grep` test.

Reply:
Hm, even on Python 3? Let me try and see what it says. Sphinx is angry about something else as well:

> docstring of scipy.sparse.csgraph.maximum_bipartite_matching:115: WARNING: Unexpected indentation.

I'm unable to build the documentation locally at the moment (running into a "ModuleNotFoundError: No module named 'scipy._lib.uarray'" when calling `make html-scipyorg`), so I might have to create a bit of noise in the PR.

Reply:
Okay, thanks for the pointer; looks like I managed to get through (and again, sorry about the PR noise/CI spam): Cythonize just really doesn't like anything that's not 7-bit ASCII (although as yourself, I was also able to build without issues locally; maybe Unicode support has improved recently?)

Adding `# -*- coding: utf-8 -*-` didn't seem to do much, but just getting rid of all non-ASCII characters and checking with what amounts to

```python
with open('_matching.pyx', 'rb') as f:
    codecs.ascii_decode(f.read())
```

did suffice. The documentation worked fine, and it's only Cython that wasn't happy, so without having looked into whether or not there's a smarter way to go about fixing this, one could split the module in two, one for Cython and one ordinary Python module with the docstring, but that seems overkill for a single ő; I'm going to leave it as is if no one complains.

Reply:
> ModuleNotFoundError: No module named `scipy._lib.uarray`

this probably indicates you're picking up a scipy version in the doc build that's not your in-place build. `_lib.uarray` was very recently added, it's not in 1.3.0

Reply:
Thanks for the hint @rgommers, and for bearing with my fumbling. I've been referring to [these docs](https://scipy.github.io/devdocs/dev/contributor/rendering_documentation.html) to get Sphinx up and running but haven't had much luck; walking through those steps on a freshly `git clone`d SciPy repository leads to the `_lib.uarray` error.

Reply:
yeah that misses one step - it assumes you've done something like `python setup.py develop` already. I.e. before `make html` or `make html-scipyorg`, make sure that `>>> import scipy` works and picks up the scipy build you are working on

Reply:
I'll file a separate issue for the missing doc step